charts/pgadmin4/templates/networkpolicy.yaml

```diff
@@ -14,4 +14,4 @@ spec:
       app.kubernetes.io/instance: {{ .Release.Name }}
   ingress:
   - ports:
-    - port: {{ .Values.service.port }}
+    - port: {{ .Values.service.targetPort }}
```

Comment:
Why are you adjusting the ingress port to targetPort 'http'?

Reply:
Not sure what exactly are you asking. The point is that network policy must be applied to container port, rather than service port, and targetPort actually represents the container port to which the service connects on the container. This was confirmed when you set a "default-deny" ingress network policy and deploy pgAdmin4. Without the change, you'll not be able to access pgAdmin4 through service port, either via ingress definition, node port or port-forward. And named ports get translated when policy is applied.

Reply:
```service:
  type: ClusterIP
  port: 80
  targetPort: http```

In all the charts I've personally used, the port of ingress always points to this specific port.

Sorry, perhaps I don't fully understand why your adjustment would be required for the default use of this repository. Could be that in your specific case it is unique and requires a different configuration?

Reply:
I use "extraContainers" to add a keycloak auth proxy. All requests made to the service should go to the keycloak port, rather than default "http" port of the postgres, and keycloak will talk to postgresql using localhost. So I have to change targetPort to port exposed by keycloak container, and in turn, to allow network policy to only connect to that port, rather than default port.

Reply:
@krezovic oke clear, thank you for your patience. Please bump the chart version and we are good to go and merge it.